manim/renderer/opengl_renderer.py

```diff
@@ -380,18 +380,27 @@ def render_mobject(self, mobject):
             mesh.render()
 
     def get_texture_id(self, path):
-        if path not in self.path_to_texture_id:
-            # A way to increase tid's sequentially
-            tid = len(self.path_to_texture_id)
-            im = Image.open(path)
-            texture = self.context.texture(
-                size=im.size,
-                components=len(im.getbands()),
-                data=im.tobytes(),
-            )
-            texture.use(location=tid)
-            self.path_to_texture_id[path] = tid
-        return self.path_to_texture_id[path]
+        if str(path) not in self.path_to_texture_id:
```

Comment:
is `path` a `Path` object? In which case I would recommend using `os.fspath`. If it's not using the `repr` function instead of `str` function is better.

Reply:
It could be; as of now, it can be 1 of 4 types: `np.ndarray`, `PIL.Image.Image`, `str`, or `pathlib.Path`. I think using `repr` is best here.